logstash-core/src/main/java/org/logstash/Logstash.java

```diff
@@ -92,16 +102,59 @@ private static void configureNashornDeprecationSwitchForJavaAbove11() {
         }
     }
 
-    private static void handleCriticalError(Throwable t, String[] errorMessage) {
-        LOGGER.error(t);
-        if (errorMessage != null) {
-            for (String err : errorMessage) {
-                System.err.println(err);
+    private static void installGlobalUncaughtExceptionHandler() {
+        Thread.setDefaultUncaughtExceptionHandler((thread, e) -> {
+            if (e instanceof Error) {
+                handleFatalError("uncaught error (in thread " + thread.getName() + ")",  e);
+            } else {
+                LOGGER.error("uncaught exception (in thread " + thread.getName() + ")", e);
             }
+        });
+    }
+
+    private static void handleFatalError(String message, Throwable t) {
+        LOGGER.fatal(message, t);
```

Comment:
previously LS never logged stack-trace for fatals, which likely is due trying hard to manually log `:backtrace` on rescues.
anyhow, having last resort proper logging is very useful (even for Ruby throwables).